Merge search filter to Iteration2

Sahadporn opened the pull request:
I add Search by keyword, search by amenities function, and UI with paginator.
Please take a look at my code.

Sahadporn committed:
add simple search

Sahadporn committed:
add search filter

Sahadporn committed:
Add [search by amenities | result page ui | Paginator]

kaesrel reviewed (commented):
Are rooms supposed to be included in the search?

Sahadporn committed:
fix minor bug in roomSet_list

Search will now display not found if you type wrong keyword. Instead of all room.

Sahadporn committed:
fix amenity search return roomSet.

kaesrel reviewed (approved):
There is no longer any problem with the search filter now.
